WalletWasabi/Blockchain/TransactionOutputs/ISmartCoin.cs

```diff
@@ -0,0 +1,16 @@
+using NBitcoin;
+
+namespace WalletWasabi.Blockchain.TransactionOutputs;
+
+public interface ISmartCoin
```

Comment:
> This is an ICoin + AnonymitySet. I mean, a NBitcoin::ICoin already has and TxOut (Amount, ScriptPubKey) and an OutPoint (TransactionId, Index).

It is not, because (as you already said yourself) `ICoin` + AnonymitySet contains `ScriptPubKey` instead of `ScriptType`. The `ScriptPubKey` itself is not needed in the coin selection algorithm.